### PR TITLE
Add export products section

### PR DESCRIPTION
**Pull request by IKarbowiak**
Add documentation about exporting products.

### CODE REVIEW

File: docs/developer/export-products.mdx [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+(see the `exportInfo` input fields). If no variant fields are provided,
+each row will contain only product fields:
+
+| id  | name         | description                    | category    | product weight |
+|-----|--------------|--------------------------------|-------------|----------------|
+| 61  | Orange Juice | No additives, no preservatives | juices      | 500 g          |
+| 65  | Apple Juice  | No additives, great taste      | juices      | 1000 g         |
+| 101 | Polo Shirt   | Nice and comfortable           | polo-shirts | 200 g          |
+
+### The charts below explains workflow of the exporting products:
```
Comment (maarcingebala) [inline docs/developer/export-products.mdx:48]: One last suggestion - titles should be concise and rather short, so I would move this sentence to a paragraph and change the title.

```suggestion
### Workflow

The charts below explain the internal workflow of exporting products:
```

File: docs/developer/export-products.mdx [hunk 2]
```diff
@@ -0,0 +1,353 @@
+---
+title: Exporting Products Information
```
Comment (maarcingebala) [inline docs/developer/export-products.mdx:2]: I think just "Exporting Products" should be fine here:

```suggestion
title: Exporting Products
```

File: docs/developer.mdx [hunk 3]
```diff
@@ -23,6 +23,7 @@ Whether you're building a custom-branded storefront or looking for ways to exten
 - [**Checkout**](developer/checkout.mdx): the checkout flow.
 - [**Users**](developer/users.mdx): creating and managing customer accounts.
 - [**Metadata**](developer/metadata.mdx): store additional information about items.
+- [**Importing and exporting**](developer/export-products.mdx): import and export data.
```
Comment (patrys) [inline docs/developer.mdx:26]: It says import and export but only one direction is currently supported.

File: docs/developer/export-products.mdx [hunk 4]
```diff
@@ -0,0 +1,355 @@
+---
+title: Exporting Products
+sidebar_label: Exporting products
```
Comment (patrys) [inline docs/developer/export-products.mdx:3]: This is only required if the sidebar label (what appears in the sidebar) is different from the title (what appears at the top of the page).

File: docs/developer/export-products.mdx [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+:::
+
+### Export file structure
+Each row in the exported file represents a single product variant, but it also contains
+general product fields.
+For example, if a product has three variants, there will be three lines in total.
+Each line will contain common product fields such as name or description,
+and fields specific to given variant:
+
+| id | name         | description                    | variant sku | variant weight |
```
Comment (patrys) [inline docs/developer/export-products.mdx:32]: Are the column titles actually lowercase?

Comment (IKarbowiak) [reply docs/developer/export-products.mdx:32]: Yes

File: docs/developer.mdx (hunk 3 above)
Comment (IKarbowiak) [reply docs/developer.mdx:26]: That's true. So maybe, for now, it should be just exporting.

Comment (patrys) [reply docs/developer.mdx:26]: Yeah, let's call it "exporting data".

**Pull request merged**